pulp_2to3_migration/app/migration.py

```diff
@@ -50,29 +53,56 @@
     """
     A coroutine to migrate pre-migrated repositories.
     """
+
+    repos_to_create = plan.get_pulp3_repository_setup().keys()
     progress_data = dict(
         message='Creating repositories in Pulp 3', code='creating.repositories', total=0
     )
     with ProgressReport(**progress_data) as pb:
         pulp2repos_qs = Pulp2Repository.objects.filter(pulp3_repository_version=None)
-        pb.total += pulp2repos_qs.count()
-        pb.save()
 
-        for pulp2repo in pulp2repos_qs:
-            # if pulp2 repo_id is too long, its hash is included in pulp3 repo name
-            pulp3_repo_name = pulp2repo.pulp2_repo_id
-            if len(pulp3_repo_name) > 255:
-                repo_name_hash = hashlib.sha256(pulp3_repo_name.encode()).hexdigest()
-                pulp3_repo_name = '{}-{}'.format(pulp3_repo_name[:190], repo_name_hash)
-
-            repo, created = Repository.objects.get_or_create(
-                name=pulp3_repo_name,
-                description=pulp2repo.pulp2_description)
-            if created:
-                pb.increment()
-            else:
-                pb.total -= 1
-                pb.save()
+        # no specific migration plan for repositories
+        if not repos_to_create:
+            pb.total += pulp2repos_qs.count()
+            pb.save()
+
+            for pulp2repo in pulp2repos_qs:
+                # if pulp2 repo_id is too long, its hash is included in pulp3 repo name
+                pulp3_repo_name = pulp2repo.pulp2_repo_id
+                if len(pulp3_repo_name) > 255:
+                    repo_name_hash = hashlib.sha256(pulp3_repo_name.encode()).hexdigest()
+                    pulp3_repo_name = '{}-{}'.format(pulp3_repo_name[:190], repo_name_hash)
+
+                repo, created = Repository.objects.get_or_create(
+                    name=pulp3_repo_name,
+                    description=pulp2repo.pulp2_description)
+                if created:
+                    pb.increment()
+                else:
+                    pb.total -= 1
+                    pb.save()
+
+        # specific migration plan for repositories
+        else:
+            pb.total += len(repos_to_create)
+            pb.save()
+
+            for pulp3_repo_name in repos_to_create:
+                try:
+                    pulp2repo = pulp2repos_qs.get(pulp2_repo_id=pulp3_repo_name)
```

Comment:
>(pulp2_repo_id=pulp3_repo_name)

This looks a little like a typo but I'm not sure -- why is the Pulp 2 repo being looked up by a Pulp 3 repo name?

Reply:
The idea is that if you preserve pulp2 name (you set in the MP to create a pulp3 repo with the same name) you likely want to move a repo as is, including its description. Maybe its a wrong assumption, I'm open for feedback.

Reply:
Ack, it's fine, I just wanted to be sure there was a reason.